pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -418,3 +426,41 @@ func constructServices(p *v1.Pod, services []model.ServiceInfo) map[string]*work
 	}
 	return res
 }
+
+func getPodLocality(ctx krt.HandlerContext, Nodes krt.Collection[*v1.Node], pod *v1.Pod) *workloadapi.Locality {
```

Comment:
In previous processing, we respect `"istio-locality"` label first, then node labels

Reply:
Yeah I saw that as legacy and wasn't sure we actually want to carry it forward in ambient?